File: common/data/src/main/java/org/thingsboard/server/common/data/id/TbResourceId.java
package org.thingsboard.server.common.data.id;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import org.thingsboard.server.common.data.EntityType;

import java.util.UUID;

public class TbResourceId extends UUIDBased implements EntityId {

    private static final long serialVersionUID = 1L;

    @JsonCreator
    public TbResourceId(@JsonProperty("id") UUID id) {
        super(id);
    }

    @ApiModelProperty(position = 2, required = true, value = "string", example = "TB_RESOURCE", allowableValues = "TB_RESOURCE")
    @Override
    public EntityType getEntityType() {
        return EntityType.TB_RESOURCE;
    }
}

File: ui-ngx/src/app/shared/models/id/tb-resource-id.ts
import { EntityId } from '@shared/models/id/entity-id';
import { EntityType } from '@shared/models/entity-type.models';

export class TbResourceId implements EntityId {
  entityType = EntityType.TB_RESOURCE;
  id: string;
  constructor(id: string) {
    this.id = id;
  }
}
